server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -128,7 +132,26 @@ public HttpInfo info() {
 
     @Override
     public HttpStats stats() {
-        return new HttpStats(httpChannels.size(), totalChannelsAccepted.get());
+        pruneClientStats(false);
+        return new HttpStats(new ArrayList<>(httpChannelStats.values()), httpChannels.size(), totalChannelsAccepted.get());
+    }
+
+    /**
+     * Prunes client stats of entries that have been disconnected for more than five minutes.
+     *
+     * @param throttled When true, executes the prune process only if more than 60 seconds has elapsed since the last execution.
+     */
+    void pruneClientStats(boolean throttled) {
+        if (throttled == false || (threadPool.relativeTimeInMillis() - lastClientStatsPruneTime > TimeUnit.SECONDS.toMillis(60))) {
```

Comment:
Can you extract this 60 second value and the 5 minute value into private static variables in this class? They're a little easier to discover and change there if we want to tweak them in the future (and it prevents the function call, even if it would end up getting inlined)